src/Milkdrop2XBMC.cpp

```diff
@@ -162,96 +145,46 @@ char* WideToUTF8( const wchar_t* WFilename )
 //-- GetPresets ---------------------------------------------------------------
 // Return a list of presets to XBMC for display
 //-----------------------------------------------------------------------------
-extern "C" unsigned int GetPresets(char ***presets)
+bool CVisualizationMilkdrop2::GetPresets(std::vector<std::string>& presets)
 {
-	if( !presets || !IsInitialized )
-	{
-		return 0;
-	}
+	if(!IsInitialized )
+		return false;
 
 	while( !g_plugin.m_bPresetListReady )
 	{
 
```

Comment:
it's waiting for the completion of downloading of preset list which happens in parallel